src/main/java/org/ideoholic/curium/model/account/service/AccountService.java

```diff
@@ -1564,138 +1564,190 @@ public VoucherPrintResponseDto viewVouchersPrint(VoucherPrintDto voucherPrintDto
-		String twoAccounts = null;
-		
-		Map<VoucherEntrytransactions,String> voucherMap = new LinkedHashMap<VoucherEntrytransactions, String>();
-		int financialYearId = new AccountDAO().getCurrentFinancialYear(Integer.parseInt(httpSession.getAttribute(BRANCHID).toString())).getFinancialid();
-		voucherTransactions = new AccountDAO().getVoucherEntryTransactions(fromDate, toDate, financialYearId, Integer.parseInt(httpSession.getAttribute(BRANCHID).toString()), voucherType);
-		
-		for (VoucherEntrytransactions voucherEntry : voucherTransactions) {
-			twoAccounts = new AccountDAO().getAccountName(voucherEntry.getDraccountid())+"--"+new AccountDAO().getAccountName(voucherEntry.getCraccountid());
-			voucherMap.put(voucherEntry, twoAccounts);
-		}
-		
-		try {
+	public ResultResponse exportVoucher(ExportVoucherDto exportVoucherDto){
+		String nextVoucher = DataUtil.emptyString(exportVoucherDto.getNextVoucher());
 
-			// Creating an excel file
-			XSSFWorkbook workbook = new XSSFWorkbook();
-			XSSFSheet sheet = workbook.createSheet("vouchertransactions");
-			Map<String, Object[]> data = new LinkedHashMap<String, Object[]>();
-			Map<String, Object[]> headerData = new HashMap<String, Object[]>();
-			headerData.put("Header",
-					new Object[] { "Voucher Transactions"});
-			Map<String, Object[]> headerData1 = new HashMap<String, Object[]>();
-			headerData1.put("Header",
-					new Object[] { "From Date: "+fromDate+"  To Date: "+toDate+""});
-			Map<String, Object[]> headerData2 = new HashMap<String, Object[]>();
-			headerData2.put("Header",
-					new Object[] { "Sl No.","Voucher No","Dr Account -- Cr Account","Narration","Amount"});
-			//"Sl.No", "Voucher No", 
-			int i = 1;
-			
-			for (Entry<VoucherEntrytransactions, String> accBal : voucherMap.entrySet()) {
-				
-				//Integer.toString(i),Integer.toString(accBal.getKey().getTransactionsid()),
-				data.put(Integer.toString(i),
-						new Object[] { Integer.toString(i),Integer.toString(accBal.getKey().getTransactionsid()),accBal.getValue().toString(),
-								accBal.getKey().getNarration(), df.format(accBal.getKey().getDramount())});
-				i++;
-				total = total.add(accBal.getKey().getDramount());
+		if(nextVoucher.equalsIgnoreCase("Receipt")){
+			ResultResponse resultResponse = exportVoucher(exportVoucherDto, 1);
+			if(resultResponse.isSuccess()){
+				exportVoucherDto.setNextVoucher(nextVoucher);
+				return resultResponse;
+				//receiptdetails
 			}
-			
-			Row headerRow = sheet.createRow(0);
-			Object[] objArrHeader = headerData.get("Header");
-			int cellnum1 = 1;
-			for (Object obj : objArrHeader) {
-				Cell cell = headerRow.createCell(cellnum1++);
-				if (obj instanceof String)
-					cell.setCellValue((String) obj);
+
+		}else if(nextVoucher.equalsIgnoreCase("Payment")){
+
+			ResultResponse resultResponse = exportVoucher(exportVoucherDto, 2);
+			if(resultResponse.isSuccess()){
+				exportVoucherDto.setNextVoucher(nextVoucher);
+				return resultResponse;
+				//receiptdetails
```

Comment:
The comment `//receiptdetails` is not needed in each of the IF condition. Originally it was only in the first IF condition.